src/jit/morph.cpp

```diff
@@ -2800,7 +2800,11 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* callNode)
         assert(argx->gtType != TYP_STRUCT);
 
         /* Increment the argument register count and argument index */
-        if (!varTypeIsFloating(argx->gtType))
+        bool fltPassedInInt = false;
+#ifdef ARM_SOFTFP
+        fltPassedInInt = true;
+#endif
+        if (!varTypeIsFloating(argx->gtType) || fltPassedInInt)
```

Comment:
I'm not sure how (or if) 'this' can be a float. I suspect this code related to float is unnecessary. Especially since there's nothing here about varargs, which won't use the float regs (for arm32)

---
In reply to: [57778119](https://github.com/dotnet/coreclr/pull/3929#discussion_r57778119) [](ancestors = 57778119)